common/reflect_resource.go

```diff
@@ -463,7 +485,11 @@ func iterFields(rv reflect.Value, path []string, s map[string]*schema.Schema,
 	fields := listAllFields(rv)
 	for _, field := range fields {
 		typeField := field.sf
-		fieldName := chooseFieldName(typeField)
+		// Always pass in an empty list for path because we unwrap the aliases map on each call.
+		// We do not use the `path` variable in this function because it sometimes contains indices
+		// and sometimes it is empty. Thus we rely on the caller of iterFields to unwrap the keys
+		// of the aliases map.
+		fieldName := chooseFieldNameWithAliases(typeField, []string{}, aliases)
```

Comment:
Why do we need both the path and aliases map here? With the alias unwrapping, we should need one or the other, not both. Wherever we recurse into a nested field, we can just unwrap the aliases one more time, so the path parameter should be unnecessary.

Reply:
Yeah you're right, I guess this is because originally I wanted to keep the same map and update the path, but it's not always possible that way with `iterFields`. Let me update this